### ISSUE TITLE
order is not correct when explicitly defining `Taylor1(Vector,order)`

### ISSUE THREAD

**Issue by blas-ko** [JuliaDiff/TaylorSeries.jl#120]
If one does
```julia
julia> using TaylorSeries;

julia> t = Taylor1([1.0,2,3,4,5],2)
 1.0 + 2.0 t + 3.0 t² + 4.0 t³ + 5.0 t⁴ + 𝒪(t³)

julia> length(t.coeffs), t.order
 (5,2)
```
notice the `𝒪(t³)` even when there are higher order terms in `t`. 

This could be treated in two ways:
+ Truncate the `Taylor1` object into the given `order`.
+ Extend the order of the `Taylor1` object into the lenght of the array of terms. 

I prefer the first solution because if you explicitly give the order to your polynomial, you would normally want to keep that order always. 

Feedback on this issue is welcomed; I can then do a PR based on what's discussed.

**Comment by lbenet** [comment]
Thanks for reporting! I think this must have been introduced in 62d8184, or somewhere around, when the code was reorganized. I'll solve this.

**Comment by lbenet** [comment]
See #122

**Comment by lbenet** [comment]
Solved by #122. Closing.
